web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -0,0 +1,414 @@
+    }
+  }
+
+  async function handleOnProceed() {
+    props.updateAgentMeta({
+      kube: confirmedCluster,
+      resourceName: confirmedCluster.name,
+    } as EksMeta);
+
+    props.nextStep();
+  }
+
+  const hasIamPermError = isIamPermError(fetchClustersAttempt);
+
+  const closeEnrollmentDialog = () => {
+    setEnrollmentState({ status: 'notStarted' });
+  };
+
+  const enrollmentNotAllowed =
+    fetchClustersAttempt.status === 'processing' ||
+    !selectedCluster ||
+    enrollmentState.status !== 'notStarted';
+
+  let command = '';
+  if (selectedCluster) {
+    command = generateCmd({
+      namespace: 'teleport-agent',
+      clusterName: selectedCluster.name,
+      proxyAddr: ctx.storeUser.state.cluster.publicURL,
+      tokenId: joinToken.id,
+      clusterVersion: ctx.storeUser.state.cluster.authVersion,
+      resourceId: joinToken.internalResourceId,
+      isEnterprise: ctx.isEnterprise,
+      isCloud: ctx.isCloud,
+      automaticUpgradesEnabled: ctx.automaticUpgradesEnabled,
+      automaticUpgradesTargetVersion: ctx.automaticUpgradesTargetVersion,
+      joinLabels: [...selectedCluster.labels, ...selectedCluster.joinLabels],
+      disableAppDiscovery: !isAppDiscoveryEnabled,
+    });
+  }
+
+  return (
+    <Box maxWidth="1000px">
+      <Header>Enroll an EKS Cluster</Header>
+      {fetchClustersAttempt.status === 'failed' && !hasIamPermError && (
+        <Danger mt={3}>{fetchClustersAttempt.statusText}</Danger>
+      )}
+      <Text mt={4}>
+        Select the AWS Region you would like to see EKS clusters for:
+      </Text>
+      <AwsRegionSelector
+        onFetch={fetchClustersWithNewRegion}
+        onRefresh={refreshClustersList}
+        clear={clear}
+        disableSelector={fetchClustersAttempt.status === 'processing'}
+      />
+      {!hasIamPermError && tableData.currRegion && (
+        <>
+          <Box mb={2}>
+            <Toggle
+              isToggled={isAppDiscoveryEnabled}
+              onToggle={() => setAppDiscoveryEnabled(!isAppDiscoveryEnabled)}
+            >
+              <Box ml={2} mr={1}>
+                Enable Kubernetes App Discovery
+              </Box>
+              <ToolTipInfo>
+                Teleport's Kubernetes App Discovery will automatically identify
+                and enroll to Teleport HTTP applications running inside a
+                Kubernetes cluster.
+              </ToolTipInfo>
+            </Toggle>
+          </Box>
+          <ClustersList
+            items={tableData.items}
+            fetchStatus={tableData.fetchStatus}
+            selectedCluster={selectedCluster}
+            onSelectCluster={setSelectedCluster}
+            fetchNextPage={fetchNextPage}
+          />
+          <StyledBox mb={5} mt={5}>
+            <Text mb={2}>Automatically enroll selected EKS cluster</Text>
+            <ButtonSecondary
+              width="215px"
+              type="submit"
+              onClick={handleOnEnroll}
```

Comment:
Nit:
```suggestion
              onClick={enroll}
```

We're not handling some kind of a "enroll" event here. I believe we want to say that clicking this button will enroll an EKS cluster, in which case the function could as well be called `enroll`.

It's not a huge deal here, but it becomes confusing whenever generic DOM event handlers are mixed with functions that are supposed to perform specific business logic.